src/test/java/com/epam/qalab/uitests/FilteringAndSearchingTest.java

```diff
@@ -0,0 +1,56 @@
+package com.epam.qalab.uitests;
+
+import com.epam.qalab.BaseTest;
+import com.epam.qalab.pageObject.pages.CategoryPage;
+import com.epam.qalab.pageObject.pages.HomePage;
+import org.testng.Assert;
+import org.testng.annotations.*;
+
+import java.util.ArrayList;
+
+public class FilteringAndSearchingTest extends BaseTest {
+
+    @BeforeMethod
+    public CategoryPage openCategoryPage() {
+        HomePage homePage = new HomePage(webDriver);
+        setUpDriver();
+        return homePage.open().choseCategory();
+    }
+
+    @AfterTest
+    public void afterTest() {
+        quit();
+    }
+
+    @Test
+    public void successfulFilteredByFeaturedBrand() {
+
+        var categoryPage = openCategoryPage();
+        String nameFeaturedBrand = categoryPage.getNameFeaturedBrand();
+        categoryPage = categoryPage
+                .selectFeaturedBrand();
+
+        Assert.assertTrue(categoryPage.getProductsDescription().stream().allMatch(item -> item.getText().contains(nameFeaturedBrand)));
+    }
+
+    @Test
+    public void successfulFilteringByPrice() {
+
+        ArrayList<Double> elementsPrices = openCategoryPage()
+                .selectPriceRangeOption()
+                .getPricesOfDisplayedElements();
+
+        Assert.assertTrue(elementsPrices.stream().allMatch(priceElem -> priceElem <= 25));
```

Comment:
The same here

Reply:
yes I know about it.. the reason is that some products have few variants with different prices. And filter take all of this prices for one product and try to match to range selected in filter. If any of them is inside range this product is displayed on list.
Maybe I should fix it but it can take more time :(

Reply:
I put comment in file with test, sometimes this test can be failed ;)